provisioner/shell/provisioner_acc_test.go

```diff
@@ -1,41 +1,81 @@
 package shell_test
 
 import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/packer/helper/tests/acc"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/hashicorp/packer/packer"
+
 	"github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/packer/command"
 	testshelper "github.com/hashicorp/packer/helper/tests"
 )
 
-func TestBuildShellProvisionerWithBuildVariablesSharing(t *testing.T) {
-	testshelper.AccTestPreValidate(t)
+func TestShellProvisioner(t *testing.T) {
+	p := os.Getenv("ACC_TEST_PROVISIONERS")
+	if p != "all" && !strings.Contains(p, "shell") {
+		t.Skip()
+	}
+	acc.TestProvisionersAgainstBuilders(new(ShellProvisionerAccTest), t)
+}
+
+type ShellProvisionerAccTest struct{}
+
+func (s *ShellProvisionerAccTest) GetName() string {
+	return "shell"
+}
+
+func (s *ShellProvisionerAccTest) GetConfig() (string, error) {
+	filePath := filepath.Join("./test-fixtures", "shell-provisioner.txt")
+	config, err := os.Open(filePath)
+	if err != nil {
+		return "", fmt.Errorf("Expected to find %s", filePath)
+	}
+	defer config.Close()
+
+	file, err := ioutil.ReadAll(config)
+	return string(file), nil
+}
+
+func (s *ShellProvisionerAccTest) GetProvisionerStore() packer.MapOfProvisioner {
+	return packer.MapOfProvisioner{
+		"shell": func() (packer.Provisioner, error) { return command.Provisioners["shell"], nil },
+		"file":  func() (packer.Provisioner, error) { return command.Provisioners["file"], nil },
+	}
+}
+
+func (s *ShellProvisionerAccTest) IsCompatible(builder string, vmOS string) bool {
+	return vmOS == "linux"
```

Comment:
@SwampDragons this should contain the provisioner validation of compatible OS and builders type. Does it resolves the case you said?